### ISSUE TITLE
Loses PR content(title, message) if unable to push to remote branch

### ISSUE THREAD

**Issue by bryceosterhaus** [node-gh/gh#768]
**Version number**
2.8.3

**Describe the bug**
Occasionally the remote branch I will be pushing to is not able to be updated unless I force push. But I don't know this until after I fill out the title and message of the pull-request. So typically I will craft a PR and finish the content, then while node-gh is trying to push it fails and loses my previous content.

**To Reproduce**

1. create a local branch that you are unable to push without force pushing
2. create a PR, fill out content
3. as the pr sends, it will fail and you will lose your content.

**Expected behavior**
I would expect it to try and push my changes to my remote before I fill out the PR message and title.
Or I would expect it to somehow save my content for my next try.

**Comment by protoEvangelion** [comment]
Thanks @bryceosterhaus so essentially you are saying we should push your commit before prompting you for PR input?

**Comment by bryceosterhaus** [comment]
@protoEvangelion yeah I think pushing to remote before adding the PR content would make sense.
